### PR TITLE
C-9526

### PR DESCRIPTION
**Pull request by ptnapoleon**
Rebase of #603 
@mambocab to review

### CODE REVIEW

File: jmx_test.py [hunk 1]
```diff
@@ -201,3 +201,34 @@ def test_compactionstats(self):
...
+        """
+        Check functioning of nodetool failuredetector.
+        @jira_ticket CASSANDRA-9526
+        """
+
+        cluster = self.cluster
+        cluster.populate(3).start(wait_for_binary_proto=True)
+        node1, node2, node3 = cluster.nodelist()
+
+        phivalues = node1.nodetool("failuredetector")[0].splitlines()
```
Comment (mambocab) [inline jmx_test.py]: It worries me a little that there's no validation of the output.

File: jmx_test.py [hunk 2]
```diff
@@ -201,3 +201,34 @@ def test_compactionstats(self):
...
+        endpoint1Values = phivalues[1].split()
+        endpoint2Values = phivalues[2].split()
+
+        endpoint1 = endpoint1Values[0][1:-1]
+        endpoint2 = endpoint2Values[0][1:-1]
+
+        self.assertTrue(endpoint1.startswith("127.0.0"))
+        self.assertTrue(endpoint2.startswith("127.0.0"))
```
Comment (mambocab) [inline jmx_test.py]: Well, OK, not _no_ validation, but this seems insufficient.

Comment (ptnapoleon) [inline jmx_test.py]: Well one of these should be 127.0.0.2, and the other should be 127.0.0.3. I think the ordering is just not guaranteed. I'll improve the validation

**Comment by mambocab** [comment]
Code looks good. I accidentally ran this with `--no-skip` and it passed on 2.2 and 2.1, which worries me.

**Comment by ptnapoleon** [comment]
Don't worry about that.

**Comment by mambocab** [comment]
Operator error on my part, not a problem

**Pull request merged**